tasks/mynt.ts

```diff
@@ -993,3 +994,57 @@ task(
       logger.warn("====================================");
     }
   });
+
+task("mynt:setMassetTokenTransferWithPermit", "set mAsset token transfer with permit address in massetManager contract")
+.setAction(async ({}, hre) => {
+  const {
+    ethers,
+    deployments: { get },
+    getNamedAccounts,
+  } = hre;
+
+  const dllrTransferWithPermitDeployment = await get("DllrTransferWithPermit");
+  const massetManager = await ethers.getContract("MassetManager")
+  const { deployer } = await getNamedAccounts();
+  const owner = await massetManager.owner();
+
+  const deployerKey = await getAuthorizedDeployerKey();
+  const multisigDeployment = await get(deployerKey["multisig"]);
+  const timelockOwnerDeployment = await get(deployerKey["sip"]["timelockOwner"]);
+  const timelockAdminDeployment = await get(deployerKey["sip"]["timelockAdmin"]);
+
+  if (owner.toLowerCase() == multisigDeployment.address.toLowerCase()) {
+    const multisigAddress = (await get("MultisigWallet")).address;
+    const data = massetManager.interface.encodeFunctionData(
+      "setMassetTokenTransferWithPermit",
+      [dllrTransferWithPermitDeployment.address]
+    );
+    await sendWithMultisig(
+      hre,
+      multisigAddress,
+      massetManager.address,
+      data,
+      deployer
+    );
+  } else if (owner.toLowerCase() == timelockOwnerDeployment.address.toLowerCase() || owner.toLowerCase() == timelockAdminDeployment.address.toLowerCase()) {
```

Comment:
pls extract this to a helper function isSip(owner) so that we had it encapsulated and could change easily if logic for SIP creation changes e.g. when deployed to Sivryn layer in the future